zone/spell_effects.cpp

```diff
@@ -6789,23 +6785,19 @@ void Mob::CheckNumHitsRemaining(NumHit type, int32 buff_slot, uint16 spell_id)
 	bool bDepleted = false;
 	int buff_max = GetMaxTotalSlots();
 
-#ifdef BOTS
 	std::string buff_name;
 	size_t buff_counter = 0;
 	bool buff_update = false;
-#endif
 
 	//Spell specific procs [Type 7,10,11]
 	if (IsValidSpell(spell_id)) {
 		for (int d = 0; d < buff_max; d++) {
 			if (buffs[d].spellid == spell_id && buffs[d].hit_number > 0 &&
 			    spells[buffs[d].spellid].hit_number_type == static_cast<int>(type)) {
 
-#ifdef BOTS
 				buff_name = spells[buffs[d].spellid].name;
```

Comment:
This seems to change how this logic fundamentally works when bots are enabled, at the very least rule gate this still or lets ask why this behaves differently when bots are enabled because it affects any mob

Reply:
Bot logic here feeds into a SendMarqueeMessage to the Client whenever the Bot is hit and have a buff with a hit counter... honestly this seems very annoying to me, and high spam.

I'd consider just removing it,.. not sure who would actually use this, worst case just lock it behind a rule.

`			bot_owner->CastToClient()->SendMarqueeMessage(
				Chat::Yellow,
				510,
				0,
				1000,
				3000,
				StringFormat(
					"%s has [%u] hit%s remaining on '%s'",
					GetCleanName(),
					buff_counter,
					(buff_counter == 1 ? "" : "s"),
					buff_name.c_str()
				)
			);`

Reply:
This is toggled with owner options, right? I agree it is a pretty pointless and annoying option but is disabled by default.

Reply:
Ahh yes, correct. the Bot logic here would be fine then.